cloudapi/v1/cloudapi.proto

```diff
@@ -6,49 +6,161 @@ import "google/protobuf/timestamp.proto";
 
 option go_package = "github.com/sourcegraph/cloud-api/go/cloudapi/v1;cloudapiv1";
 
-message ListInstancesRequest {}
+message Instance {
+  string id = 1;
+  optional InstanceDetails instance_details = 2;
+  optional PlatformDetails platform_details = 3;
+  optional InstanceState instance_state = 4;
+}
+
+enum InstanceType {
+  INSTANCE_TYPE_UNSPECIFIED = 0;
+  INSTANCE_TYPE_INTERNAL = 1;
+  INSTANCE_TYPE_TRIAL = 2;
+  INSTANCE_TYPE_PRODUCTION = 3;
+  INSTANCE_TYPE_EPHEMERAL = 4;
+}
+
+message InstanceDetails {
+  string name = 1;
+  InstanceType instance_type = 2;
+  string environment = 3;
+  string version = 4;
+  optional string hostname = 5;
+  optional string admin_email = 6;
+  // instance_features requires infrastructure changes i.e. enable cody, enable nat, ephemeral instance
+  map<string, string> instance_features = 7;
+  // application_features
+  map<string, string> application_features = 8;
```

Comment:
what are `application_defauts`? maybe show some example here

Reply:
Agreed with @michaellzc to remove `application_features` - in tge begining people will experiment with features via modyfing site config, b/c we cannot predict how complicated feature configs might be.